src/libmongoc/tests/test-mongoc-client-session.c

```diff
@@ -757,6 +756,27 @@ test_end_sessions_pooled (void *ctx)
    _test_end_sessions (true);
 }
 
+/* Sends ping to server via client_session.  useful for marking
+ * server_sessions as used so that they are pushed back to the session pool */
+static void
+send_ping (mongoc_client_t *client, mongoc_client_session_t *client_session)
+{
+   bson_t ping_cmd = BSON_INITIALIZER;
+   bson_error_t error;
+   bool ret;
+
+   BCON_APPEND (&ping_cmd, "ping", BCON_INT32 (1));
+
+   ret = mongoc_client_command_with_opts (client,
+                                          "admin",
+                                          &ping_cmd,
+                                          NULL,
+                                          &client_session->server_session->lsid,
```

Comment:
Instead of using the `lsid` document directly (which is private), append the session into the options like so:

```
bson_t opts = BSON_INITIALIZER;
mongoc_client_session_append (client_session, &opts);
ret = mongoc_client_command_with_opts (client, ... &opts, NULL /* reply */, error);
```

Note, `mongoc_client_session_append` appends the id associated with the `mongoc_client_session_t`, which libmongoc uses to look up the `mongoc_client_session_t`. It does not append the `lsid` of the server session directly, so the behavior will differ.

Reply:
See also: https://github.com/mongodb/mongo-c-driver/blob/1.16.0/src/libmongoc/tests/test-mongoc-client-session.c#L633-L640 for an example.